third_party/sl4a/src/main/java/com/google/android/mobly/snippet/rpc/SimpleServer.java

```diff
@@ -138,17 +138,19 @@ private InetAddress getPrivateInetAddress() throws UnknownHostException, SocketE
 
         InetAddress candidate = null;
         Enumeration<NetworkInterface> nets = NetworkInterface.getNetworkInterfaces();
-        for (NetworkInterface netint : Collections.list(nets)) {
-            if (!netint.isLoopback() || !netint.isUp()) { // Ignore if localhost or not active
-                continue;
-            }
-            Enumeration<InetAddress> addresses = netint.getInetAddresses();
-            for (InetAddress address : Collections.list(addresses)) {
-                if (address instanceof Inet4Address) {
-                    Log.d("local address " + address);
-                    return address; // Prefer ipv4
+        if (nets != null) {
```

Comment:
instead of adding nesting, could we check for `nets == null` instead?

Also, is it possible to add simple unit tests for this

Reply:
+1, prefer checking for `nets == null` instead